coreneuron/io/phase2.cpp

```diff
@@ -0,0 +1,1211 @@
+
+    nodecounts = std::vector<int>(nodecounts_, nodecounts_ + n_mech);
+    delete[] nodecounts_;
+
+    v_parent_index.resize(n_node);
+    actual_a.resize(n_node);
+    actual_b.resize(n_node);
+    actual_area.resize(n_node);
+    actual_v.resize(n_node);
+    if (n_diam > 0) {
+        actual_diam.resize(n_node);
+    }
+    int* v_parent_index_ = const_cast<int*>(v_parent_index.data());
+    double* actual_a_ = const_cast<double*>(actual_a.data());
+    double* actual_b_ = const_cast<double*>(actual_b.data());
+    double* actual_area_ = const_cast<double*>(actual_area.data());
+    double* actual_v_ = const_cast<double*>(actual_v.data());
+    double* actual_diam_ = const_cast<double*>(actual_diam.data());
+    (*nrn2core_get_dat2_2_)(thread_id, v_parent_index_, actual_a_, actual_b_,
+            actual_area_, actual_v_, actual_diam_);
+
+    tmls.resize(n_mech);
+
+    auto& param_sizes = corenrn.get_prop_param_size();
+    auto& dparam_sizes = corenrn.get_prop_dparam_size();
+    int dsz_inst = 0;
+    for (size_t i = 0; i < n_mech; ++i) {
+        auto& tml = tmls[i];
+        int type = types[i];
+
+        tml.nodeindices.resize(nodecounts[i]);
+        tml.data.resize(nodecounts[i] * param_sizes[type]);
+        tml.pdata.resize(nodecounts[i] * dparam_sizes[type]);
+
+        int* nodeindices_ = nullptr;
+        double* data_ = const_cast<double*>(tml.data.data());
+        int* pdata_ = const_cast<int*>(tml.pdata.data());
+        (*nrn2core_get_dat2_mech_)(thread_id, i, dparam_sizes[type] > 0 ? dsz_inst : 0, nodeindices_, data_, pdata_);
+        if (dparam_sizes[type] > 0)
+            dsz_inst++;
+        std::copy(nodeindices_, nodeindices_ + nodecounts[i], tml.nodeindices.data());
+        free_memory(nodeindices_);
+    }
+
+    int* output_vindex_ = nullptr;
+    double* output_threshold_ = nullptr;
+    int* pnttype_ = nullptr;
+    int* pntindex_ = nullptr;
+    double* weight_ = nullptr;
+    double* delay_ = nullptr;
+    (*nrn2core_get_dat2_3_)(thread_id, n_weight, output_vindex_, output_threshold_, pnttype_,
+            pntindex_, weight_, delay_);
+
+    output_vindex = std::vector<int>(output_vindex_, output_vindex_ + nt.n_presyn);
+    delete[] output_vindex_;
+
+    output_threshold = std::vector<double>(output_threshold_, output_threshold_ + n_real_output);
+    delete[] output_threshold_;
+
+    int n_netcon = nt.n_netcon - nrn_setup_extracon;
+    pnttype = std::vector<int>(pnttype_, pnttype_ + n_netcon);
+    delete[] pnttype_;
+
+    pntindex = std::vector<int>(pntindex_, pntindex_ + n_netcon);
+    delete[] pntindex_;
+
+    weights = std::vector<double>(weight_, weight_ + n_weight);
+    delete[] weight_;
+
+    delay = std::vector<double>(delay_, delay_ + n_netcon);
+    delete[] delay_;
+
+    (*nrn2core_get_dat2_corepointer_)(nt.id, npnt);
+
+    for (size_t i = 0; i < n_mech; ++i) {
+        if (!corenrn.get_bbcore_read()[types[i]]) {
+            continue; // I don't get this test
```

Comment:
something for @pramodk to comment on.

Reply:
It seems we don't call it, in fact, it's why I don't get what this tests **means**

Something like: I can call `nrn2core_*` function because `get_bbcore_read` is defined?

Reply:
If MOD file has BBCOREPOINTER then we have to get extra serialised data. This is what achieved via:
```
(*nrn2core_get_dat2_corepointer_mech_)(nt.id, tmls[i].type, icnt, dcnt, iArray_, dArray_);
```